src/erlang_ls_server.erl

```diff
@@ -126,92 +121,18 @@ handle_request(Socket, Request) ->
       send_notification(Socket, M, P)
   end.
 
+%% @doc Dispatch the handling of the method to erlang_ls_protocol_impl
 -spec handle_method(binary(), map()) ->
+  {response, map() | null} | {} | {notification, binary(), map()}.
+handle_method(Method, Params) ->
+  Function = method_to_function_name(Method),
+  try erlang_ls_protocol_impl:Function(Params)
```

Comment:
Nice, thanks. FYI I am working on a branch which will end up slightly modifying things around this area. The idea is to define a "provider" behaviour that different providers will implement. Providers will then be responsible for handling different sections of the protocol.

Reply:
Do all providers have the same "interface"?